### PR TITLE
docs/PORTING.md: Update APIs needed for porting MCUboot

### PR DESCRIPTION
**Pull request by rajivr**
Update the list of APIs needed for porting MCUboot to a new platform.

### CODE REVIEW

File: docs/PORTING.md [hunk 1]
```diff
@@ -117,12 +117,18 @@ int     flash_area_write(const struct flash_area *, uint32_t off,
...
-int     flash_area_to_sectors(int idx, int *cnt, struct flash_area *ret);
+/*< What is value is read from erased flash bytes. */
+uint8_t flash_area_erased_val(const struct flash_area *);
+/*< Reads len bytes from off, and checks if the read data is erased. */
+int     flash_area_read_is_empty(const struct flash_area *fa, uint32_t off,
+                     void *dst, uint32_t len);
+/*< Given flash area ID, return info about sectors within the area. */
+int     flash_area_get_sectors(int fa_id, uint32_t *count,
+                     struct flash_sector *sectors);
 /*< Returns the `fa_id` for slot, where slot is 0 (primary) or 1 (secondary) */
```
Comment (utzig) [inline docs/PORTING.md]: Please also mention the `image_index` parameter

File: docs/PORTING.md [hunk 2]
```diff
@@ -117,12 +117,18 @@ int     flash_area_write(const struct flash_area *, uint32_t off,
...
-int     flash_area_id_from_image_slot(int slot);
+int     flash_area_id_from_multi_image_slot(int image_index, int slot);
 /*< Returns the slot, for the `fa_id` supplied */
```
Comment (utzig) [inline docs/PORTING.md]: Comment should have `area_id` (betteer) or parameter should be renamed `fa_id`. Please mention `image_index` in the comment.

**Comment by rajivr** [comment]
Pushed new revision with changes. Please review.

File: docs/PORTING.md [hunk 3]
```diff
@@ -117,12 +117,22 @@ int     flash_area_write(const struct flash_area *, uint32_t off,
...
 uint8_t flash_area_align(const struct flash_area *);
-/*< Initializes an array of flash_area elements for the slot's sectors */
-int     flash_area_to_sectors(int idx, int *cnt, struct flash_area *ret);
-/*< Returns the `fa_id` for slot, where slot is 0 (primary) or 1 (secondary) */
-int     flash_area_id_from_image_slot(int slot);
-/*< Returns the slot, for the `fa_id` supplied */
-int     flash_area_id_to_image_slot(int area_id);
+/*< What is value is read from erased flash bytes. */
+uint8_t flash_area_erased_val(const struct flash_area *);
+/*< Reads len bytes from off, and checks if the read data is erased. */
```
Comment (utzig) [inline docs/PORTING.md]: Sorry, have missed this one, but would be nice to also mention here that this function should return `1` for empty, `0` for not empty, and < 0 for errors.

**Comment by rajivr** [comment]
Pushed revision with suggested changes for `flash_area_read_is_empty`. Please review.

**Pull request merged**